include/cache.php

```diff
@@ -46,43 +50,71 @@ public function switch_blog( $new_blog_id ) {
 	}
 
 	/**
-	 * Add a value in cache.
+	 * Adds a value in cache.
 	 *
 	 * @since 1.7
+	 * @since 3.6 Returns the cached value.
 	 *
 	 * @param string $key  Cache key.
 	 * @param mixed  $data The value to add to the cache.
-	 * @return void
+	 * @return mixed
+	 *
+	 * @phpstan-param non-empty-string $key
+	 * @phpstan-param TCacheData $data
+	 * @phpstan-return TCacheData
 	 */
 	public function set( $key, $data ) {
 		$this->cache[ $this->blog_id ][ $key ] = $data;
+		return $data;
 	}
 
 	/**
-	 * Get value from cache.
+	 * Returns value from cache.
 	 *
 	 * @since 1.7
 	 *
 	 * @param string $key Cache key.
 	 * @return mixed
+	 *
+	 * @phpstan-param non-empty-string $key
+	 * @phpstan-return TCacheData|false
 	 */
 	public function get( $key ) {
 		return isset( $this->cache[ $this->blog_id ][ $key ] ) ? $this->cache[ $this->blog_id ][ $key ] : false;
 	}
 
 	/**
-	 * Clean the cache (for this blog only).
+	 * Cleans the cache (for this blog only).
 	 *
 	 * @since 1.7
 	 *
-	 * @param string $key Cache key.
+	 * @param string $key Optional. Cache key. An empty string to clean the whole cache for the current blog.
+	 *                    Default is an empty string.
 	 * @return void
 	 */
 	public function clean( $key = '' ) {
-		if ( empty( $key ) ) {
+		if ( '' === $key ) {
 			unset( $this->cache[ $this->blog_id ] );
 		} else {
 			unset( $this->cache[ $this->blog_id ][ $key ] );
 		}
 	}
+
+	/**
+	 * Generates and returns a "unique" cache key, depending on `$data` and prefixed by `$prefix`.
+	 *
+	 * @since 3.6
+	 *
+	 * @param string              $prefix String to prefix the cache key.
+	 * @param string|array|object $data   Data.
+	 * @return string
+	 *
+	 * @phpstan-param non-empty-string $prefix
+	 * @phpstan-return non-empty-string
+	 */
+	public function get_unique_key( string $prefix, $data ): string {
+		$data = maybe_serialize( $data );
+		$data = is_scalar( $data ) ? (string) $data : '';
+		return $prefix . md5( $data );
```

Comment:
The check is useless as maybe_serialize() always returns a scalar.
```suggestion
		/** @var scalar **/
		$serialized = maybe_serialize( $data );
		return $prefix . md5( (string) $serialized );
```